Automattic/newspack-plugin
# feat(ras): add toggle to force checkout registration

### All Submissions:






### Changes proposed in this Pull Request:



Closes https://app.asana.com/0/1205234045751551/1208650122332813/f

The changes in this PR require the following Blocks PR to be checked out: https://github.com/Automattic/newspack-blocks/pull/1931

This PR adds a toggle to disable/enable pre checkout auth:
![Screenshot 2024-10-29 at 16 32 01](https://github.com/user-attachments/assets/273fecce-7657-4ec6-87da-aa0d926fe6e7)

### How to test the changes in this Pull Request:

Be sure to also checkout the following blocks PR before testing: https://github.com/Automattic/newspack-blocks/pull/1931

1. As admin go to Newspack > Engagement > Show Advanced Settings and enable forced checkout registration in the Modal Checkout Login and Registration section pictured above
2. As a reader, go through the checkout flow via any checkout or donation button
3. Confirm you are required to sign in or register a new account before completing checkout
4. As admin again, disable forced checkout registration via the same menu as step 1
5. As a reader, go through checkout again
6. This time, confirm you are not required to sign in or register before completing checkout
7. Also confirm a new account was created and you were automatically logged in after checkout if using an email address not already associated with an account on the test site.

### Other information:
